lib/statsig.ex

```diff
@@ -1,241 +1,72 @@
-    user = Utils.get_user_with_env(user, get_tier(server))
+  def check_gate(user, gate) do
+    user = Statsig.Utils.get_user_with_env(user)
 
-    result = Statsig.Evaluator.eval(user, gate, :gate, server)
-    log_exposures(server, user, result.exposures, :gate)
+    result = Statsig.Evaluator.eval(user, gate, :gate)
+    log_exposures(user, result.exposures, :gate)
 
     case result do
       %{reason: :not_found} -> {:error, :not_found}
       _ -> {:ok, result.result}
     end
   end
-  # Make a layer module
-  # The layer module can take a layer struct and return a value
-  # use atoms as keys
-
-  def get_config(user, config, server \\ __MODULE__)
-  def get_config(nil, _config, _server), do: {:error, :no_user}
 
-  def get_config(user, config, server) do
-    user = Utils.get_user_with_env(user, get_tier(server))
-    result = Statsig.Evaluator.eval(user, config, :config, server)
-    log_exposures(server, user, result.exposures, :config)
+  def get_config(user, config) do
+    user = Statsig.Utils.get_user_with_env(user)
+    result = Statsig.Evaluator.eval(user, config, :config)
+    log_exposures(user, result.exposures, :config)
 
-    # TODO - could probably hand back a Result struct
     case result do
       %{reason: :not_found} -> {:error, :not_found}
-      # TODO - this be {:ok, result}
       _ -> %{rule_id: Map.get(result.rule, "id"), value: result.value}
     end
   end
 
-  def get_experiment(user, exp), do: get_config(user, exp)
-  def get_experiment(user, exp, server), do: get_config(user, exp, server)
-
+  def get_experiment(user, experiment) do
+    get_config(user, experiment)
+  end
 
-  def log_event(event, server \\ __MODULE__) do
-    GenServer.call(server, {:log, event})
+  def log_event(event) do
+    GenServer.cast(Statsig.Logging, {:log_event, event})
   end
 
-  def log_event(user, event_name, value, metadata, server \\ __MODULE__) do
-    user = Utils.get_user_with_env(user, get_tier(server))
+  def log_event(user, event_name, value, metadata) do
+    user = Statsig.Utils.get_user_with_env(user)
     event = %{
       "eventName" => event_name,
       "value" => value,
       "metadata" => metadata,
       "time" => :os.system_time(:millisecond),
-      "user" => Utils.sanitize_user(user)
+      "user" => Statsig.Utils.sanitize_user(user)
     }
-    GenServer.call(server, {:log, event})
+    GenServer.cast(Statsig.Logging, {:log_event, event})
   end
 
-
-  def state(server \\ __MODULE__), do: GenServer.call(server, :state)
-
-  # TODO - pipeline this
-  def lookup(name, type, server \\ __MODULE__), do: :ets.lookup(ets_name(server), {name, type})
-
-  # TODO - pipeline this
-  def all(type, server \\ __MODULE__),
-    do: :ets.match(ets_name(server), {{:"$1", type}, :_}) |> List.flatten()
-
-  # TODO - will break if the server is a PID
-  def ets_name(server) do
-    [server, :statsig_store]
-    |> Enum.join("_")
-    |> String.replace(".", "_")
-    |> String.to_atom()
+  def shutdown() do
```

Comment:
this is a bit overloaded, and actually managing application lifecycle should be done via elixir methods that already exist.  I think ill replace this in favor of flush()